File: sdk-playwrigth/pages/metamask/swap_confirm.go
package metamask

import (
	"fmt"

	"github.com/playwright-community/playwright-go"
)

type SwapConfirm struct {
	page playwright.Page
}

func NewSwapConfirm(page playwright.Page) *SwapConfirm {
	return &SwapConfirm{
		page: page,
	}
}

func (s *SwapConfirm) Confirm() error {
	err := s.page.Locator(`//*[@id="app-content"]/div/div[2]/div/div/div[3]/div/button[2]`).Click()
	if err != nil {
		return fmt.Errorf("клик кнопки: %w", err)
	}
	return nil
}

File: tests/ui/pages/metamask/SwapConfirm.ts
import { expect, Page } from "@playwright/test";
import { BasePage } from "../BasePage";

export class MetamaskSwapConfirm extends BasePage {
  constructor(page: Page) {
    super(page);
    this.page = page;
  }

  private SELECTORS = {
    CONFIRM_BUTTON: this.page.locator('//*[@id="app-content"]/div/div[2]/div/div/div[3]/div/button[2]')
  }

  async confirmClick() {
    await this.SELECTORS.CONFIRM_BUTTON.click();
  }

  async navigateTo() {
    await this.page.goto(this.url);
  }

  async verifyRequiredElementsPresent() {
    var timeout = 10*1000
    await expect(this.SELECTORS.CONFIRM_BUTTON).toBeVisible({timeout:timeout})
  }
}
